tedge/src/cli/config/commands/get.rs

```diff
@@ -0,0 +1,33 @@
+use crate::cli::config::ConfigKey;
+use crate::command::{Command, ExecutionContext};
+
+pub struct GetConfigCommand {
+    pub config_key: ConfigKey,
+    pub config: tedge_config::TEdgeConfig,
+}
+
+impl Command for GetConfigCommand {
+    fn description(&self) -> String {
+        format!(
+            "get the configuration value for key: {}",
+            self.config_key.key
+        )
+    }
+
+    fn execute(&self, _context: &ExecutionContext) -> Result<(), anyhow::Error> {
+        match (self.config_key.get)(&self.config) {
```

Comment:
Yes, function application. If you write `self.config_key.get(&self.config)` it will call the `get` method of `config_key`, which does not exist. `self.config_key.get` is really a function closure. There is a `call` method on trait `Fn` (see https://doc.rust-lang.org/std/ops/trait.Fn.html) but it's nightly only. So in the future we can write:

```rust
match self.config_key.get.call(&self.config) {
   ...
}```